src/data/set/intervals/interval.lean

```diff
@@ -0,0 +1,154 @@
+/-
+Copyright (c) 2020 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou
+-/
+
+import data.set.intervals
+
+/-!
+# Intervals
+
+`a..b` is the set of elements lying between `a` and `b`, with both endpoints included.
+
+`Icc a b` requires the assumption `a ≤ b` to be meaningful, which is sometimes inconvenient. This
+version of interval is always the set of things lying between `a` and `b`, regardless of the
+relative order of `a` and `b`.
+-/
+
+universe u
+
+namespace set
+
+section decidable_linear_order
+
+variables {α : Type u} [decidable_linear_order α] {a a₁ a₂ b b₁ b₂ x : α}
+
+/-- `interval a b` is the set of elements lying between `a` and `b`, with `a` and `b` included. -/
+def interval (a b : α) := if a ≤ b then Icc a b else Icc b a
+
+localized "infix `..`:1025 := interval" in interval
+
+@[simp] lemma interval_of_le (h : a ≤ b) : a..b = Icc a b :=
+if_pos h
+
+@[simp] lemma interval_of_le' (h : b ≤ a) : a..b = Icc b a :=
+by { rw interval, split_ifs with h', { rw le_antisymm h h' }, refl }
+
+lemma interval_swap (a b : α) : a..b = b..a :=
+or.elim (le_total a b) (by simp {contextual := tt}) (by simp {contextual := tt})
+
+lemma interval_of_lt (h : a < b) : a..b = Icc a b :=
+interval_of_le (le_of_lt h)
+
+lemma interval_of_lt' (h : b < a) : a..b = Icc b a :=
+interval_of_le' (le_of_lt h)
+
+lemma interval_of_not_le (h : ¬ a ≤ b) : a..b = Icc b a :=
+if_neg h
+
+lemma interval_of_not_le' (h : ¬ b ≤ a) : a..b = Icc a b :=
+by { rw interval_swap, exact if_neg h }
+
+@[simp] lemma interval_self : a..a = {a} :=
+set.ext $ by simp [le_antisymm_iff, and_comm]
+
+@[simp] lemma nonempty_interval : set.nonempty a..b :=
+by { rw interval, split_ifs, { simp [h] }, simp [le_of_not_le h] }
+
+@[simp] lemma left_mem_interval : a ∈ a..b :=
+by { rw interval, split_ifs with h, { rwa left_mem_Icc }, simp [le_of_not_le h] }
+
+@[simp] lemma right_mem_interval : b ∈ a..b :=
+by { rw interval_swap, exact left_mem_interval }
+
+lemma mem_interval_of_mem_Icc (h : x ∈ Icc a b) : x ∈ a..b :=
+by { rwa interval_of_le, exact le_trans h.1 h.2 }
+
+lemma mem_interval_of_mem_Icc' (h : x ∈ Icc b a) : x ∈ a..b :=
+by { rw interval_swap, exact mem_interval_of_mem_Icc h }
+
+lemma mem_interval_of_le (ha : a ≤ x) (hb : x ≤ b) : x ∈ a..b :=
+mem_interval_of_mem_Icc ⟨ha, hb⟩
+
+lemma mem_interval_of_le' (hb : b ≤ x) (ha : x ≤ a) : x ∈ a..b :=
+mem_interval_of_mem_Icc' ⟨hb, ha⟩
+
+lemma interval_subset_interval (h₁ : a₁ ∈ a₂..b₂) (h₂ : b₁ ∈ a₂..b₂) : a₁..b₁ ⊆ a₂..b₂ :=
```

Comment:
With `min .. max` definition this would be something like `Icc_subset_Icc (le_min h1.1 h2.1) (max_le h1.2 h2.2)`.